airflow/providers/amazon/aws/hooks/ecs.py

```diff
@@ -198,7 +197,12 @@ def _event_to_str(self, event: dict) -> str:
         return f"[{formatted_event_dt}] {message}"
 
     def get_last_log_messages(self, number_messages) -> list:
-        return [log["message"] for log in deque(self._get_log_events(), maxlen=number_messages)]
+        last_logs_iterator = self.logs_hook.get_log_events(
+            self.log_group, self.log_stream_name, start_from_head=False
+        )
+        truncated = list(itertools.islice(last_logs_iterator, number_messages))
+        # need to reverse the order to put the logs back in order after getting them from the end
+        return [log["message"] for log in reversed(truncated)]
```

Comment:
```suggestion
        truncated = itertools.islice(last_logs_iterator, number_messages)
        messages = [log["message"] for log in truncated]
        # need to reverse the order to put the logs back in order after getting them from the end
        return messages.reverse()
```

I wonder if this would be slightly more efficient (one less list allocation). Probably marginal either way.

Reply:
hmm, possibly yes. Hard to say :p
However, reverse has no return value (it reverses in place), so we cannot return it directly ;)